### PR TITLE
feat: Add sections to overview.html

### PR DESCRIPTION
**Pull request by DafyddLlyr**
image

Follows on from https://github.com/theopensystemslab/planx-new/pull/1527 in order to complete ticket https://trello.com/c/R9b4hKmo/2298-99-add-section-markers-to-overviewhtml-layout

Not done much on styling here, but the card pattern here has been suggested as an option - https://design-system.service.gov.uk/components/summary-list/

### CODE REVIEW

File: data/example.json [hunk 1]
```diff
@@ -498,22 +498,18 @@
       "responses": [{ "value": "Yes" }]
     },
     {
-      "metadata": "",
```
Comment (DafyddLlyr) [inline data/example.json:501]: I removed this as it seemed like an issue with the example - I can't recreate this when uploading images?

File: src/Main.tsx [hunk 2]
```diff
@@ -2,7 +2,8 @@ import React from "react";
 import ReactDOM from "react-dom/client";
 import { SubmissionOverviewDocument } from "./overview/SubmissionOverview";
 import { BoundaryMapDocument } from "./map/BoundaryMapDocument";
-import example from "../data/example.json";
+// import example from "../data/example.json";
```
Comment (builditben) [inline src/Main.tsx:5]: nit: does this comment need to stay?

Comment (DafyddLlyr) [reply src/Main.tsx:5]: Fixed! This was just an oversight 👍

**Comment by DafyddLlyr** [comment]
I'll merge this after https://github.com/theopensystemslab/planx-document-templates/pull/13 as there will likely be conflicts that will be easier / quicker to fix on this smaller PR.

**Comment by jessicamcinchak** [comment]
@DafyddLlyr just merged #13 this morning after yesterday's thumbs-up at planning - thanks for waiting to merge this one! Please shout if I can help with any sections rebasing. 

One thing I'd expect to show up here is needing to possibly handle/filter out "empty" sections in the "Proposal details" sections of the overview, because the relevant data has already been pulled out into custom formatted sections further up the doc.

**Comment by DafyddLlyr** [comment]
@jessicamcinchak This is good for a review whenever 👌

File: src/overview/OverviewDocument.tsx [hunk 3]
```diff
@@ -125,37 +126,56 @@ function Boundary(props: { data: PlanXExportData[] }): JSX.Element {
   );
 }
 
-function ProposalDetails(props: { data: PlanXExportData[] }): JSX.Element {
+function ProposalDetails(props: { data: PlanXExportData[], title?: string }): JSX.Element {
   return (
     <Box>
-      <h2>Proposal details</h2>
+      <h2>{props.title || "Proposal details"}</h2>
```
Comment (jessicamcinchak) [inline src/overview/OverviewDocument.tsx:132]: :+1: this makes sense!

**Review by jessicamcinchak — APPROVED** [review]
looks good! thanks for adding overview & overview with sections separately to the example artifacts, makes it extra easy to compare and troubleshoot in future :+1:

**Pull request merged**